hazelcast/src/test/java/com/hazelcast/executor/SmallClusterTest.java

```diff
@@ -309,4 +311,47 @@ public Integer call() throws Exception {
             return ++state;
         }
     }
+
+    @Test
+    public void submitToAllMembers_NonSerializableResponse() {
```

Comment:
I would not test this low-level functionality using a service implementation, but use SPI directly, to make test cleaner and to avoid surprises.

Reply:
For example, see following tests:

```kotlin
@Test
fun invocation_shouldFail_whenResponse_isNotSerializable() {
    val factory = createHazelcastInstanceFactory()
    val hz1 = factory.newHazelcastInstance()
    val hz2 = factory.newHazelcastInstance()

    val operationService = HazelcastTestSupport.getOperationService(hz1)
    val target = HazelcastTestSupport.getAddress(hz2)

    val future = operationService
            .invokeOnTarget<Any>(null, NonSerializableResponseOperation(), target)

    try {
        future.join()
    } catch (e: HazelcastSerializationException) {
    }
}

@Test
fun invocation_shouldFail_whenNormalResponse_isNotSerializable() {
    val factory = createHazelcastInstanceFactory()
    val hz1 = factory.newHazelcastInstance()
    val hz2 = factory.newHazelcastInstance()

    val operationService = HazelcastTestSupport.getOperationService(hz1)
    val target = HazelcastTestSupport.getAddress(hz2)

    val future = operationService
            .invokeOnTarget<Any>(null, NonSerializableResponseOperation_withNormalResponseWrapper(), target)

    try {
        future.join()
    } catch (e: HazelcastSerializationException) {
    }
}

class NonSerializableResponse {}

class NonSerializableResponseOperation : Operation() {
    override fun getResponse(): Any {
        return NonSerializableResponse()
    }
}

class NonSerializableResponseOperation_withNormalResponseWrapper : Operation() {
    override fun getResponse(): Any {
        return NormalResponse(NonSerializableResponse(), callId, 0, false)
    }
}
```

Reply:
@mdogan Thanks for the tests.